packages/drawer/rc/index.js

```diff
@@ -80,47 +81,19 @@ const Body = React.forwardRef(({ children, ...rest }, forwardedRef) => {
   )
 })
 
-const drawerReducer = (state, action) => {
-  return action.type === Drawer.types.toggleDrawer
-    ? { open: !state.open }
-    : action.type === Drawer.types.closeDrawer
-    ? { open: false }
-    : action.type === Drawer.types.openDrawer
-    ? { open: true }
-    : (() => {
-        throw new Error(`Unhandled type: ${action.type}`)
-      })()
-}
-
-export const useDrawer = ({ reducer = (s, a) => a.changes, init } = {}) => {
-  const [{ open }, dispatch] = useReducer(
-    (state, action) => {
-      const changes = drawerReducer(state, action)
-      return reducer(state, { ...action, changes })
-    },
-    { open: false },
-    init
+export const DrawerProvider = ({ children, open, onToggle }) => {
+  const value = useMemo(() => ({ open, onToggle }), [open, onToggle])
+  return (
+    <DrawerContext.Provider value={value}>{children}</DrawerContext.Provider>
   )
-  const onToggle = () => {
-    dispatch({ type: Drawer.types.toggleDrawer })
-  }
-  const onOpen = () => {
-    dispatch({ type: Drawer.types.openDrawer })
-  }
-  const onClose = () => {
-    dispatch({ type: Drawer.types.closeDrawer })
-  }
-  return { open, onToggle, onClose, onOpen }
 }
 
-export const Drawer = ({ children, open, onToggle }) => {
-  const uncontrolled = useDrawer()
-  const value = React.useMemo(
-    () => ({ ...(open !== undefined ? { open, onToggle } : uncontrolled) }),
-    [open, onToggle, uncontrolled]
-  )
+export const Drawer = ({ children }) => {
+  const [open, setOpen] = useState(false)
   return (
-    <DrawerContext.Provider value={value}>{children}</DrawerContext.Provider>
+    <DrawerProvider open={open} onToggle={() => setOpen(!open)}>
```

Comment:
Not the best name but it occurred to me while the stateReducer patterns might be good for something like Autocomplete/Typeahead and DateInput where we have numerous microinteractions it's probably not the best for single microinteraction components like Drawer or toggle. We've talked about doing more composability and building lower level components. That why I like this pattern for simpler interactive components where a developer may want to take over control.